crates/revm/src/cached.rs

```diff
@@ -152,6 +166,221 @@ impl<DB: DatabaseRef> Database for CachedReadsDbMut<'_, DB> {
     }
 }
 
+impl<DB: AccountReader> AccountReader for CachedReadsDbMut<'_, DB> {
```

Comment:
~~do we need these?~~

ah we need these to satisfy the stateprovider trait?

but why do we need this traits now?